README.md

```diff
@@ -125,25 +134,50 @@ Sometimes you need a service to be available alongside your Jupyter server. In t
 
 A service is described by 
 
-* **template** - a name of a template imported in OpenShift which describes how to deploy the service
-* **parameters** - an object containing key-value pairs to be processed into the template
+* **resources** - these are the names of configMaps which contain various templates. For each resource there can be a different configMap with a different path. Or we can use a single configMap containing all templates and these are simply taken by using the correct path.
+* **configuration** - an object containing key-value pairs to be processed into the template
+* **labels** - object containing labels that will be added into the template
 * **return** - an object containing key-value pairs where value is a [JSON Path](https://github.com/kennknowles/python-jsonpath-rw) walkable in the uploaded ConfigMap/CustomResource and key is a name of environment variable under which the value will be available in Jupyterhub Singleuser server
 
 A `USERNAME` parameter is automatically added based on user's name to separate services by user.
 
+As long as the resource is correctly defined in the template and there is a custom resource definition on the cluster, the template can be used to create any custom resource, or a configMap.
+
+The following example shows a ConfigMap containing `sparkClusterTemplate` object which can be used as a `path` reference in the `service` definition.
 ```
-template: jupyterhub-spark-operator-configmap
-parameters:
-  WORKER_NODES: 2
-  MASTER_NODES: 1
-  MEMORY: 2Gi
-  CPU: 2
-  SPARK_IMAGE: jkremser/openshift-spark:2.3-latest
-return:
-  SPARK_CLUSTER: "metadata.name"
-```
+kind: ConfigMap
+apiVersion: v1
+metadata:
+  name: spark-cluster-template
+data:
+  sparkClusterTemplate: |
+    kind: SparkCluster
+    apiVersion: radanalytics.io/v1
+    metadata:
+      name: spark-{{ name }}-cluster
```

Comment:
To keep this same as it used to be in previous version, can you make this `spark-cluster-{{ user }}`?